packages/os/os.spec

```diff
@@ -121,6 +121,11 @@ Summary: Bottlerocket userdata configuration system
 
 %package -n %{_cross_os}netdog
 Summary: Bottlerocket network configuration helper
+%if %{with systemd_networkd}
+Requires: %{_cross_os}systemd-networkd
+%else
+Requires: %{_cross_os}wicked
+%endif
```

Comment:
Why the move from `release` to `os`? Whenever there is a "dog" here that has a dependency on a third-party-package, `os/Cargo.toml` is updated with a comment as well as the spec, but the package is still a dependency in `release.spec`. See `kexec-tools` and `prairiedog` for reference

Reply:
You can still use `%{with systemd_networkd}` from the `release.spec` file (unless somebody in the team disagrees).

Reply:
> You can still use `%{with systemd_networkd}` from the `release.spec` file (unless somebody in the team disagrees).

You're right, I could do that.  I was trying to avoid proliferating additional places the macro was used  we would need to clean up later.

Reply:
I would rather add the condition to `os.spec` - which is already heavily conditionalized - rather than making `release.spec` variant-specific.

Reply:
Sounds good. There is still a missing comment in `os/Cargo.toml` referencing the new "dependency" :+1: